source/crc.c

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include "checksums.h"
+
+#include "aws/checksums/crc.h"
+#include "aws/common/byte_buf.h"
+
+PyObject *aws_py_checksums_crc32(PyObject *self, PyObject *args) {
+    (void)self;
+
+    Py_buffer input_buf;
```

Comment:
Py_buffer needs to be freed when you're done with it. See how cpython did it:
https://github.com/python/cpython/blob/ff230153fd2b1d0576df21ae6c29f0f2fdb0244f/Modules/clinic/zlibmodule.c.h#L775